tests/tracking/test_tracking.py

```diff
@@ -590,6 +590,114 @@ def test_log_figure_raises_error_for_unsupported_figure_object_type():
         mlflow.log_figure("not_figure", "figure.png")
 
 
+@pytest.mark.large
+@pytest.mark.parametrize("subdir", [None, ".", "dir", "dir1/dir2", "dir/.."])
+def test_log_image_numpy(subdir):
+    import numpy as np
+
+    filename = "image.png"
+    artifact_file = filename if subdir is None else posixpath.join(subdir, filename)
+
+    image = np.random.randint(0, 256, size=(100, 100, 3), dtype=np.uint8)
+
+    with mlflow.start_run():
+        mlflow.log_image(image, artifact_file)
+
+        artifact_path = None if subdir is None else posixpath.normpath(subdir)
+        artifact_uri = mlflow.get_artifact_uri(artifact_path)
+        run_artifact_dir = local_file_uri_to_path(artifact_uri)
+        assert os.listdir(run_artifact_dir) == [filename]
```

Comment:
Could we also enhance these test to load back the image and verify that the contents match the original? I wonder if Pillow makes it easy to load the image back and compare to the original etc